src/main/java/net/minecraftforge/client/renderer/LevelRendererAdapter.java

```diff
@@ -0,0 +1,100 @@
+package net.minecraftforge.client.renderer;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.mojang.math.Matrix4f;
+import net.minecraft.client.Camera;
+import net.minecraft.client.renderer.LevelRenderer;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.util.profiling.ProfilerFiller;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.common.ForgeConfig;
+
+import java.util.Collection;
+
+/**
+ * Adapter class used to dispatch draw calls for vanilla and user-provided static
+ * render types during level rendering.
+ */
+public final class LevelRendererAdapter
+{
+    private static final LevelRendererAdapter INSTANCE = new LevelRendererAdapter();
+
+    public static LevelRendererAdapter getInstance()
+    {
+        return INSTANCE;
+    }
+
+    private LevelRendererAdapter()
+    {
+    }
+
+    /**
+     * Invoked by the renderer when it is time to render the blocks which are solid.
+     *
+     * @return {@code true} when the renderer should not render the blocks, {@code false} otherwise.
+     */
+    public boolean onRenderSolids(
+      final LevelRenderer levelRenderer,
+      final PoseStack poseStack,
+      final Camera camera,
+      final Matrix4f projectionMatrix
+    ) {
+        if (!ForgeConfig.CLIENT.useLevelRendererAdapter.get())
+            return false;
+
+        doRenderPhases(LevelRenderPhaseManager.getInstance().getSolidPhases(), camera, levelRenderer, poseStack, projectionMatrix);
+
+        return true;
+    }
+
+    /**
+     * Invoked by the renderer when it is time to render the blocks which are translucent.
+     *
+     * @return {@code true} when the renderer should not render the blocks, {@code false} otherwise.
+     */
+    public boolean onRenderTranslucent(
+      final LevelRenderer levelRenderer,
+      final PoseStack poseStack,
+      final Camera camera,
+      final Matrix4f projectionMatrix
+    ) {
```

Comment:
Formatting: braces on newlines:
```suggestion
    )
    {
```